integration_tests/src/main/python/parquet_pyarrow_test.py

```diff
@@ -0,0 +1,258 @@
+    :param parquet_path:  parquet file path
+    :return: Spark Row List
+    """
+    # By default, pyarrow read timestamp column with precision ns instead of us,
+    # Spark only supports us,
+    # coerce_int96_timestamp_unit is used to read GPU generated timestamp column as us
+    # refer to: https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html
+    # In pyspark: INT96 timestamps will be inferred as timestamps in nanoseconds
+    #
+    pyarrow_table = pa_pq.read_table(parquet_path, coerce_int96_timestamp_unit="us")
+    # convert pyarrow table to CPU result
+    return convert_pyarrow_table_to_cpu_object(pyarrow_table)
+
+def read_parquet(data_path):
+    """
+    (Fetches a function that) Reads Parquet from the specified `data_path`.
+    If the plugin is enabled, the read is done via Spark APIs, through the plugin.
+    If the plugin is disabled, the data is read via `pyarrow`.
+    :param data_path: Location of the (single) Parquet input file.
+    :return: A function that reads Parquet, via the plugin or `pyarrow`.
+    """
+
+    def read_with_pyarrow_or_plugin(spark):
+        plugin_enabled = spark.conf.get("spark.rapids.sql.enabled", "false") == "true"
+        if plugin_enabled:
+            # read on GPU
+            return spark.read.parquet(data_path)
+        else:
+            # read on pyarrow
+            rows = read_on_pyarrow(data_path)
+            # get the schema
+            schema = spark.read.parquet(data_path).schema
+            # create data frame to wrap Row list
+            # should pass schema here to avoid complex type inference error 
+            return spark.createDataFrame(rows, schema=schema)
+
+    return read_with_pyarrow_or_plugin
+
+def assert_gpu_and_pyarrow_are_compatible(base_write_path, gen_list, conf={}, pyarrow_write_conf={}):
+    """
+    Test scenarios:
+      Write by pyarrow, test readings on pyarrow and GPU
+      Write by GPU, test readings on pyarrow and GPU
+    The data is collected back to the driver and compared here, so be
+    careful about the amount of data returned.
+    """
+    pyarrow_parquet_path = base_write_path + "/pyarrow"
+    gpu_parquet_path = base_write_path + "/gpu"
+
+    def write_on_pyarrow():
+        pa_table = gen_pyarrow_table(gen_list)
+        pa_pq.write_table(pa_table, pyarrow_parquet_path, **pyarrow_write_conf)
+
+    def write_on_gpu():
+        with_gpu_session(lambda spark: gen_df(spark, gen_list).coalesce(1).write.parquet(gpu_parquet_path), conf=conf)
+
+    # write on pyarrow, compare reads between pyarrow and GPU
+    write_on_pyarrow()
+    assert_gpu_and_cpu_are_equal_collect(read_parquet(pyarrow_parquet_path), conf=conf)
+
+    # write on GPU, compare reads between pyarrow and GPU
+    write_on_gpu()
+    assert_gpu_and_cpu_are_equal_collect(read_parquet(gpu_parquet_path), conf=conf)
+
+
+# types for test_parquet_read_round_trip_for_pyarrow
+sub_gens = all_basic_gens_no_null + [decimal_gen_64bit, decimal_gen_128bit]
```

Comment:
nit: can you document why no_nulls is being used here? I assume it has something to do with NaNs instead of nulls.

Reply:
This PR is mainly comparing the diffs between Pyarrow and GPU.
Spark NullType is not supported by GPU Parquet, so skipped this type to compare.
If you think we should compare Pyarrow and Spark CPU for the NullType, then I'll file a follow-up issue.

>I assume it has something to do with NaNs instead of nulls.
No. Has nothing to to with NaNs.  The cases already tested NaNs in float types.